commit 4647731b1fb5dfeb68d9f7b54653c050e52e48e6
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sat Nov 28 17:48:46 2015 +0100

    [akonadi] Make build of testing tools possible without BUILD_TESTING

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6ee9cbe..7e8df8c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,8 @@ else()
     set(KF5Akonadi_DATA_DIR "${CMAKE_INSTALL_PREFIX}/${KDE_INSTALL_DATADIR_KF5}/akonadi")
 endif()
 
+option(BUILD_TOOLS "Build and install tools for testing purposes." TRUE)
+
 ########### Find packages ###########
 find_package(KF5ItemViews ${KF5_VERSION} CONFIG REQUIRED)
 find_package(KF5KIO ${KF5_VERSION} CONFIG REQUIRED)
@@ -93,7 +95,14 @@ add_definitions("-DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII")
 configure_file(akonadiprivate_export.h.in "${CMAKE_CURRENT_BINARY_DIR}/akonadiprivate_export.h")
 
 add_subdirectory(src)
-if(BUILD_TESTING)
+# add testrunner (application for managing a self-contained test
+# environment)
+if(BUILD_TOOLS)
+   add_subdirectory(autotests/testrunner)
+   add_subdirectory(autotests/testresource)
+   add_subdirectory(autotests/testsearchplugin)
+endif()
+if(BUILD_TOOLS AND BUILD_TESTING)
    add_subdirectory(autotests)
    add_subdirectory(tests)
 endif()
diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index 37dc10f..dc69a26 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -25,13 +25,6 @@ include_directories(
   ${Boost_INCLUDE_DIR}
 )
 
-# add testrunner (application for managing a self-contained test
-# environment)
-
-add_subdirectory(testrunner)
-add_subdirectory(testresource)
-add_subdirectory(testsearchplugin)
-
 # convenience macro to add akonadi qtestlib unit-tests
 macro(add_akonadi_test _source)
   set(_test ${_source})
